### PR TITLE
MAINT: file sink to support Event model

### LINKED ISSUE #606: FileSink to use Event model
**Is your feature request related to a problem? Please describe.**
Follow up on Event model change #538 to migrate the [FileSink](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java). 

**Describe the solution you'd like**
This sink should implement should utilized events: `... implements Sink<Record<Event>> {`

**Additional context**
This plugin will move to only `Event`'s when the `Record`is fully deprecated in v2.0

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR modifies FileSink to support Event model while maintaining backward compatibility until we fully migrated all plugins to Event model.
 
### Issues Resolved
Resolves #606 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java [hunk 1]
```diff
@@ -56,18 +57,31 @@ public FileSink(final String outputFile) {
...
+            for (final Record<Object> record : records) {
+                checkTypeAndWriteObject(record.getData(), writer);
             }
         } catch (IOException ex) {
             throw new RuntimeException(format("Encountered exception opening/creating file %s", outputFilePath), ex);
         }
     }
 
+    // Temporary function to support both trace and log ingestion pipelines.
+    // TODO: This function should be removed with the completion of: https://github.com/opensearch-project/data-prepper/issues/546
```
Comment (dlvenable) [inline data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java:72]: Please add a task to the description of #546 to note that this needs to be done in order to complete that issue.

One task noting both this class and the unit test class should be sufficient.

Comment (chenqi0805) [reply data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java:72]: #758

**Review by dlvenable — APPROVED** [review]

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java [hunk 2]
```diff
@@ -56,18 +57,31 @@ public FileSink(final String outputFile) {
...
+    private void checkTypeAndWriteObject(final Object object, final BufferedWriter writer) throws IOException {
+        if (object instanceof String) {
+            writer.write((String) object);
+            writer.newLine();
+        } else if (object instanceof Event) {
+            writer.write(((Event) object).toJsonString());
+            writer.newLine();
+        } else {
+            throw new RuntimeException("Invalid record type. FileSink only supports String and Events");
```
Comment (sbayer55) [inline data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java:81]: Is the best response to raise an exception here?

1. We could still publish the object after converting it to a string. The result may not be great but would alert the customer something is not working as expected.
2. Throwing the exception here would prevent any other (potentially good) records in the list from being written to the sink.

**Review by sbayer55 — CHANGES_REQUESTED** [review]

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java (hunk 2 above)
Comment (chenqi0805) [reply data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java:81]: @sbayer55 Good point. L81 is supposed to be transient as eventually we will only support Event type, i.e. change from Sink<Record<Object>> into Sink<Record<Event>>. Despite that, logging the error will be a better option than throwing a RuntimeException.

> We could still publish the object after converting it to a string.
The idea of specifying Record type is to unify the format. So we should avoid publishing malformatted object.

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java [hunk 3]
```diff
@@ -78,7 +81,7 @@ private void checkTypeAndWriteObject(final Object object, final BufferedWriter w
             writer.write(((Event) object).toJsonString());
             writer.newLine();
         } else {
-            throw new RuntimeException("Invalid record type. FileSink only supports String and Events");
+            LOG.error("Invalid record type. FileSink only supports String and Events");
```
Comment (dlvenable) [inline data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java:84]: We should probably log the type. But, I'm fine accepting this.

**Review by dlvenable — APPROVED** [review]

**Review by sbayer55 — APPROVED** [review]

**Pull request merged**